tests/BaseElementTest.php

```diff
@@ -236,4 +247,55 @@ public function testOnBeforeWriteNoParent()
 
         $this->assertEquals(0, (int) $element1->Sort);
     }
+
+    public function getElementCMSLinkDataProvider()
+    {
+        return [
+            'element1' => [
+                TestElement::class,
+                'elementDataObject1',
+                'http://localhost/test/1/ItemEditForm/field/ElementalArea/item/4/edit',
+                true
+            ],
+            'element2' => [
+                TestElementDataObject::class,
+                'testElementDataObject1',
+                'http://localhost/test/1/ItemEditForm/field/ElementalArea/item/5/edit',
+            ],
+            'element3' => [
+                ElementContent::class,
+                'contentDataObject1',
+                'http://localhost/test/1/ItemEditForm',
+            ],
+            'element4' => [
+                ElementContent::class,
+                'content1',
+                'http://localhost/admin/pages/edit/show/1',
+            ],
+            'element5' => [
+                ElementContent::class,
+                'contentDataObject2',
+                null
+            ],
+        ];
+    }
+
+    /**
+     * @param string $class
+     * @param string $element
+     * @param string|null $link
+     * @dataProvider getElementCMSLinkDataProvider
+     */
+
+    public function testGetElementCMSLink($class, $element, $link, $directLink = false)
+    {
+        $obj = $this->objFromFixture($class, $element);
+        
+        $class = new ReflectionClass($class);
+        $method = $class->getMethod('getElementCMSLink');
```

Comment:
Update this method to test `CMSEditLink()` instead, since ultimately that's the method that matters. We could remove this private method in the future, but `CMSEditLink()` must continue working as expected regardless.

Reply:
Also since this is testing the logic that handles both `SiteTree` and non-`SiteTree` instances, make sure there's at least one `ElementalArea` with a `SiteTree` owner in this test - and confirm that you're testing both inline and non-inline editable for `SiteTree` and non-`SiteTree` alike.